### PR TITLE
464 - Edit Trainee Personal Details & Education

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/XuX9khLQ/464-edit-trainee-personal-details-education

### Changes proposed in this pull request

- Add the personal details and education view when editing the trainee record
- Implement a helper method to allow extending core layouts, this is used to create an additional layout for the trainee record
 so that the secondary tab navigation can be re-used across each view of information.
- Integrate the TabNavigation component to navigate different views
- Implement delete function in the degree summary card to handle deletion of individual degrees

delete

### Guidance to review

- Visit https://dfe-twd-rttd-pr-229.herokuapp.com/trainees/1/edit
- Assert the trainee details and personal details/education summaries are displayed correctly
- Assert you can delete a degree

### CODE REVIEW

**Comment by darokel** [comment]
[[S] Edit Trainee Personal Details & Education](https://trello.com/c/XuX9khLQ/464-edit-trainee-personal-details-education)

File: app/components/trainees/confirmation/degrees/view.html.erb [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-    <%= render_component SummaryCard::View.new(
-        title: degree_title(degree),
-        heading_level: 2,
-        rows: get_degree_rows(degree)
-      )
-    %>
+    <%= render_component SummaryCard::View.new(title: degree_title(degree), heading_level: 2, rows: get_degree_rows(degree)) do |component| %>
+      <% if show_delete_button %>
+        <% component.with(:header_actions) do %>
+          <%= form_with model: [trainee, degree], method: :delete, local: true, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
```
Comment (defong) [inline app/components/trainees/confirmation/degrees/view.html.erb:6]: Is there a way to set this `GOVUKDesignSystemFormBuilder::FormBuilder` like 

https://github.com/DFE-Digital/register-trainee-teachers/blob/b0acf6df79d2954e26851ec6336bd9d9df0a1f67/app/controllers/application_controller.rb#L9

File: spec/support/page_objects/sections/summaries/degree_detail.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+module PageObjects
+  module Sections
+    module Summaries
+      class DegreeDetail < PageObjects::Sections::Base
+        element :type, ".govuk-summary-list__row.degree-type .govuk-summary-list__value"
+        element :subject, ".govuk-summary-list__row.subject .govuk-summary-list__value"
+        element :institution, ".govuk-summary-list__row.institution .govuk-summary-list__value"
+        element :graduation_year, ".govuk-summary-list__row.graduation-year .govuk-summary-list__value"
+        element :gradie, ".govuk-summary-list__row.grade .govuk-summary-list__value"
```
Comment (defong) [inline spec/support/page_objects/sections/summaries/degree_detail.rb:13]:
```suggestion
        element :grade, ".govuk-summary-list__row.grade .govuk-summary-list__value"
```

Comment (darokel) [reply spec/support/page_objects/sections/summaries/degree_detail.rb]: done https://github.com/DFE-Digital/register-trainee-teachers/pull/229/commits/fb4e55ecc56e03c36a11f80f0eb8b21d80888075

File: app/views/trainees/personal_details/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  <%= render_component Trainees::Confirmation::Diversity::View.new(trainee: @trainee) %>
+</div>
+
+<h2 class="govuk-heading-m">Education</h2>
+
+<h3 class="govuk-heading-s">Degrees</h3>
+
+<div class="degree-details">
+  <%= render_component Trainees::Confirmation::Degrees::View.new(trainee: @trainee, show_delete_button: true) %>
+</div>
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/views/trainees/personal_details/show.html.erb]: done https://github.com/DFE-Digital/register-trainee-teachers/pull/229/commits/fb4e55ecc56e03c36a11f80f0eb8b21d80888075

File: app/views/layouts/trainee_record.html.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    </h1>
+  </div>
+
+  <%= render_component TabNavigation::View.new(items: [
+    { name: "Training details", url: edit_trainee_path(@trainee) },
+    { name: "Personal details and education", url: trainee_personal_details_path(@trainee) },
+  ]) %>
+
+  <%= yield %>
+<% end %>
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/views/layouts/trainee_record.html.erb]: done https://github.com/DFE-Digital/register-trainee-teachers/pull/229/commits/fb4e55ecc56e03c36a11f80f0eb8b21d80888075

File: app/components/trainees/confirmation/degrees/view.html.erb (hunk 1 above)
Comment (darokel) [reply app/components/trainees/confirmation/degrees/view.html.erb]: i've just used the native approach similar to one of the other components https://github.com/DFE-Digital/register-trainee-teachers/pull/229/commits/fb4e55ecc56e03c36a11f80f0eb8b21d80888075

**Review by defong — APPROVED** [review]

**Pull request merged**